src/dopri.ts

```diff
@@ -0,0 +1,149 @@
+import * as dopri5 from "./dopri5/stepper";
+import * as interpolator from "./interpolator";
+import {RhsFn, Stepper} from "./types";
+import * as utils from "./utils";
+
+// needed for ES5 - will be ~= Number.EPSILON in ES6
+const DBL_EPSILON = Math.pow(2, -52); // = 2.220446049250313e-16
+const STEP_FACTOR_MIN = 1e-4;
+
+export function integrate(rhs: RhsFn, y: number[],
+                          t0: number, t1: number) {
+    const solver = new Dopri(rhs, y.length);
+    solver.initialise(t0, y);
+    return solver.run(t1);
+}
+
+function integrationError(message: string, t: number) {
+    return new Error(`Integration failure: ${message} at ${t}`);
+}
+
+export class Dopri {
+    public stepper: Stepper;
+    public t: number = 0.0;
+    public h: number = 0.0;
+
+    // state
+    public nSteps: number = 0;
+    public nStepsAccepted: number = 0;
+    public nStepsRejected: number = 0;
+
+    public stiffNStiff: number = 0;
+    public stiffNNonstiff: number = 0;
+    public lastError: number = 0;
```

Comment:
Do these state properties need to be public? They look more like private variables to me; presumably you wouldn't want anyone to modify these outside of the class?

Reply:
(nb if private might want to go with the `_name` naming convention for the sake of anyone using the compiled js, where there is no such thing as a true private variable)

Reply:
I made these all public when going through and just de-linting (since that was what was happening by default without the `public` keyword I believe). But you're totally right - I don't think this makes a lot of sense.  I'll have a go at tidying this interface up a bunch